app/commands.py

```diff
@@ -1411,6 +1376,31 @@ async def rmpriv(ctx: Context) -> Optional[str]:
     return f"Updated {t}'s privileges."
 
 
+@command(Privileges.DEVELOPER, hidden=True)
+async def givedonator(ctx: Context) -> Optional[str]:
+    """Gives donator to a specified player (by name) for a specified amount of time, such as '3h5m'."""
+    if len(ctx.args) < 2:
+        return "Invalid syntax: !givedonator <name> <duration>"
+
+    if not (t := await app.state.sessions.players.from_cache_or_sql(name=ctx.args[0])):
+        return "Could not find user."
+
+    if not (timespan := timeparse("".join(ctx.args[1:]))):
```

Comment:
should be just ctx.args[1] - `len(ctx.args) < 2` is asserted above

Reply:
that only checks if theres less than 2 args lol

Reply:
Oh right, didn't realize pytimeparse supported that
```py
>>> pytimeparse.parse(' 3h 5m')
11100
```

in this case, just make sure we're being consistent across all commands doing time parsing with this - if we allow spaces in one, we should allow spaces in all

Reply:
i'd have to move the delay at the end on the shutdown command then

Reply:
ah sorry shutdown is okay as-is then, just worry about user-facing commands, dev commands are lower prio

Reply:
what about silence though

Reply:
do i just move duration to the end?

Reply:
ill just leave it for now